airbyte-integrations/connectors/source-sendgrid/source_sendgrid/streams.py

```diff
@@ -196,6 +198,34 @@ def initial_path() -> str:
         return "templates"
 
 
+class Messages(SendgridStream, SendgridStreamIncrementalMixin):
+    """
+    https://docs.sendgrid.com/api-reference/e-mail-activity/filter-all-messages
+    """
+
+    data_field = "messages"
+    cursor_field = "last_event_time"
+    primary_key = "msg_id"
+    limit = 1000
+
+    def request_params(self, stream_state: Mapping[str, Any], **kwargs) -> MutableMapping[str, Any]:
+        time_filter_template = "%Y-%m-%dT%H:%M:%SZ"
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        if type(params["start_time"]) == int:
```

Comment:
I would suggest making sure that the `start_time` stored in the state is consistently an `int` by converting the `latest_benchmark` value to an `int` timestamp if the API returns a string for the `cursor` value. It will make the implementation easier and less prone to error. You can check the implementation in `source-twilio`. In `source-twilio` be consistently store ISO timestamp in the state but perform stream-specific conversion to the right format in `request_params`. If you follow this direction the timestamp format management will be easier.